clang/lib/Frontend/TextDiagnostic.cpp:1254: suspicious loop ?

Static analyser cppcheck says:

clang/lib/Frontend/TextDiagnostic.cpp:1254:40: error: When I==Spelling.size(), Spelling[I] is out of bounds. [stlOutOfBounds]

Source code is

    for (unsigned I = 0; I <= Spelling.size(); ++I) {
      // This line is done.
      if (isVerticalWhitespace(Spelling[I]) || I == Spelling.size()) {

The for loop line looks like it executes once too many times. Maybe better code

    for (unsigned I = 0; I < Spelling.size(); ++I) {

--- a/clang/lib/Frontend/TextDiagnostic.cpp
+++ b/clang/lib/Frontend/TextDiagnostic.cpp
@@ -1251,7 +1251,7 @@ highlightLines(StringRef FileData, unsigned StartLineNumber,
     unsigned LineLength = 0;
     for (unsigned I = 0; I <= Spelling.size(); ++I) {
       // This line is done.
-      if (isVerticalWhitespace(Spelling[I]) || I == Spelling.size()) {
+      if (I == Spelling.size() || isVerticalWhitespace(Spelling[I])) {
         SmallVector<TextDiagnostic::StyleRange> &LineRanges =
             SnippetRanges[L - StartLineNumber];
 

